### ISSUE TITLE
Peak does not return next event

### ISSUE THREAD

**Issue by BernieWhite** [aaubry/YamlDotNet#381]
Peak<T> nomenclature and comment indicates that it should return the next event without consuming it.

Actual behaviour is to return the current event without consuming it.

https://github.com/aaubry/YamlDotNet/blob/2c6d52253cdd8d67256bbac6b26322ffcdeb49f0/YamlDotNet/Core/ParserExtensions.cs#L87-L99

Accept indicates it checks the current event.

https://github.com/aaubry/YamlDotNet/blob/2c6d52253cdd8d67256bbac6b26322ffcdeb49f0/YamlDotNet/Core/ParserExtensions.cs#L52-L67

This is from master currently.

**Comment by aaubry** [comment]
I agree that the documentation is not very strict, but the behavior seems correct. It is not very clear what you are trying to achieve. It seems that you would want to peek on the next two events. I am correct ?
